ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -0,0 +1,49 @@
+import requests
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+
+_GCE_METADATA_URL = (
+    "http://metadata.google.internal/computeMetadata/v1?recursive=true"
+)
+_GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    # This call will throw if we aren't currently in a GCE instance
+    try:
+        all_metadata = requests.get(
+            _GCE_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
```

Comment:
We can probably defer the details of the GCP detector to a later PR, but keep in mind that the "GCE Metadata Server" is available for all GCP compute services, not just GCE. Each service may return a slightly different data structure. Also, some services pass information in environment variables. Mapping this out is probably a job for the GCP spec.

Reply:
Renamed _GCE_METADATA_URL -> _GCP_METADATA_URL to make it clear its general.

Yeah so I plan on adding support for other Google tools later, this PR is already pretty big. Yeah other tools will need to be mapped out alongside the mapping for GCE (which is present in the GCP spec already)